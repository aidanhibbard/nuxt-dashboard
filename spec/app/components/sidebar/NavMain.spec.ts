import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import NavMain from '@/components/navigation/sidebar/NavMain.vue'

describe('components/navigation/sidebar/NavMain.vue', () => {
  it('renders nav items and subitems', () => {
    const wrapper = mount(NavMain, {
      props: {
        items: [
          {
            title: 'Playground',
            url: '#',
            isActive: true,
            items: [
              { title: 'History', url: '#' },
              { title: 'Settings', url: '#' },
            ],
          },
          {
            title: 'Models',
            url: '#',
            items: [
              { title: 'Genesis', url: '#' },
            ],
          },
        ],
      },
    })

    expect(wrapper.text()).toContain('Platform')
    expect(wrapper.text()).toContain('Playground')
    expect(wrapper.text()).toContain('Models')
    expect(wrapper.text()).toContain('History')
    expect(wrapper.text()).toContain('Settings')
    expect(wrapper.text()).toContain('Genesis')
  })

  it('matches snapshot', () => {
    const wrapper = mount(NavMain, {
      props: { items: [{ title: 'Playground', url: '#', isActive: true, items: [{ title: 'History', url: '#' }] }] },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
