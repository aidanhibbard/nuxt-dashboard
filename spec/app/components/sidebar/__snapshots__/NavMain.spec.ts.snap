// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/navigation/sidebar/NavMain.vue > matches snapshot 1`] = `
"<div>
  <div>Platform</div>
  <ul>
    <div as-child="" default-open="true" class="group/collapsible">
      <li>
        <div as-child=""><button tooltip="Playground">
            <!--v-if--><span>Playground</span><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide ml-auto transition-transform duration-200 group-data-[state=open]/collapsible:rotate-90 lucide-chevron-right-icon lucide-chevron-right ml-auto transition-transform duration-200 group-data-[state=open]/collapsible:rotate-90">
              <path d="m9 18 6-6-6-6"></path>
            </svg>
          </button></div>
        <div>
          <ul>
            <li><button as-child=""><a href="#"><span>History</span></a></button></li>
          </ul>
        </div>
      </li>
    </div>
  </ul>
</div>"
`;
